<div class = "alert  alert-danger"> 
    
Attention: **veillez à bien sauvegarder votre travail** dans le bon dossier du disque réseau (dossier document) avec le bon nom (et l'extension *.ipynb*), **sinon toutes les modifications seront perdues!**

Pour reprendre votre travail, il suffit d'ouvrir le fichier .ipynb en cliquant sur *Fichier ouvrir*
</div>

# 1b. Compléments : Afficher de l'information
Dans ce notebook nous apporterons quelques compléments aux notions abordées dans le premier *notebook*

Les programmes ne se limitent pas à la réalisation de graphiques. En fait, pendant longtemps, le seul moyen d'interagir avec un programme était par le biais du texte.

Par conséquent, les programmes sont également capables de manipuler du texte, y compris de lire, d'écrire et d'afficher des textes sur un écran. Dans ce contexte, les commandes `from turtle import *` et `done()`, habituellement utilisées pour le dessin avec la tortue, deviennent inutiles. En terme informatique, les textes sont appelés des chaînes de caractères.
<pre>


</pre>
## L'instruction Print
La première instruction que nous allons voir est la fonction print qui permet d'afficher du texte à l'écran.

En Python, l'instruction `print("blabla")` affiche le texte `blabla` dans la console à droite. 

<div class = "alert  alert-warning"> 

**Important: les parenthèses et les guillemets sont nécessaires.**

</div>

<h3 style="color:teal;background-color:azure;" > <i class="fa fa-pencil" aria-hidden="true"> </i> &nbsp; Exercice 1 </h3>

Teste le programme suivant en appuyant sur Ctrl + Enter

In [1]:
print("blabla")

blabla


<h3 style="color:teal;background-color:azure;" > <i class="fa fa-pencil" aria-hidden="true"> </i> &nbsp; Exercice 2 </h3>

Écris un programme qui permet d'afficher le texte `Vive l'instruction print !`, puis vérifie en appuyant sur Ctrl + Enter.

In [3]:
# à compléter

### Afficher plusieurs lignes de texte
Pour afficher plusieurs lignes, il suffit de répéter l'instruction `print` avec un notre message.
Ainsi, le code suivant :

```python
print("blabla")
print("blibli")
```

affichera une ligne ` blabla ` puis une ligne ` blibli`.


<h3 style="color:teal;background-color:azure;" > <i class="fa fa-pencil" aria-hidden="true"> </i> &nbsp; Exercice 3 </h3>

Complète le programme pour afficher pous afficher le texte sur trois lignes suivant:
```
Bonjour !
Comment allez-vous ?
Il fait beau aujourd'hui.
```

In [ ]:
print("Bonjour !")

### Des programmes qui calculent

La fonction `print` ne sert pas qu'à afficher du texte, elle permet également d'afficher le résultat d'expressions, notamment des opérations arithmétiques.

Par exemple, `print(3+3)` affichera le résultat de l'opération, soit... `6`. 

<div class = "alert  alert-warning"> 

**Important: Pour effectuer le calcul, il faut ôter les guillemets, sinon le programme affichera le l'expression et non sont résultat. Par exemple, l'instruction `print("3+3")` afficherait simplement `3+3` et pas `6`**

</div>

<h3 style="color:teal;background-color:azure;" > <i class="fa fa-pencil" aria-hidden="true"> </i> &nbsp; Exercice 4 </h3>

Teste le code ci-dessous puis modifie-le afin d'afficher le résultat de **18975 + 98756** dans la console, et vérifie en appuyant sur Ctrl + Enter.

In [ ]:
print(3+3)

### Les opérations avec les nombres

En programmation `Python`, on a les opérateurs mathématiques suivant : 
*  \+ pour  l'addition
* \- pour la soustraction
* \*  pour la multiplication
* \\ pour la division
* \*\* pour la puissance
* \\\\ pour la division entière
* \% pour le reste de la division entière

<h3 style="color:teal;background-color:azure;" > <i class="fa fa-pencil" aria-hidden="true"> </i> &nbsp; Exercice 5 </h3>

Teste tous ces opérateurs pour bien comprendre les derniers, puis écris un code pour afficher le résultat de $(264 · 53)/22+274$ dans la console, et vérifie en appuyant sur Ctrl + Enter.

10192


<h3 style="color:teal;background-color:azure;" > <i class="fa fa-pencil" aria-hidden="true"> </i> &nbsp; Exercice 6 </h3>

Note sur une feuille de papier ce qui va être affiché par chacune des instructions suivantes:

    1. print(3+5)
    2. print(7*2)
    3. print(8/4)
    4. print(5/2)
    5. print(5//2)
    6. print(2**3)
    7. print(12%5)


<details>
<summary style="border-left:3px solid #3c763d; border-radius:2pt; width:100%; color:#3c763d; padding:6px; background-color: #dff0d8"> 
Réponses
</summary>  

<div style="border-left:3px solid #3c763d; border-radius:2pt; color:#3c763d; padding:6px; background-color: #eff0e8">

1. `8`<br>
2. `14`<br>
3. `2`<br>
4. `2.5`<br>
5. `2`<br>
6. `8`<br>
7. `2`<br>
    
</div>
</details>

<h3 style="color:teal;background-color:azure;" > <i class="fa fa-pencil" aria-hidden="true"> </i> &nbsp; Exercice 7 </h3>

Quelle opération pourrait-on utiliser pour reconnaître si un nombre est pair ou impair? (une opération qui donnerait toujours le même résultat si le nombre est pair et toujours le même résultat si le nombre est impair)


<details>
<summary style="border-left:3px solid #3c763d; border-radius:2pt; width:100%; color:#3c763d; padding:6px; background-color: #dff0d8"> 
Réponse
</summary>  

<div style="border-left:3px solid #3c763d; border-radius:2pt; color:#3c763d; padding:6px; background-color: #eff0e8">

On utilise le reste de la division entière (opérateur `%` par 2. Si le résultat est 0 le nombre est pair, si il vaut 1 c'est qu'il est impair. 
    
</div>
</details>

<h3 style="color:teal;background-color:azure;" > <i class="fa fa-pencil" aria-hidden="true"> </i> &nbsp; Exercice 8 </h3>

Note sur une feuille de papier ce qui va être affiché par chacune des instructions suivantes:

    1. print(98145 % 2)
    2. print(7841592 % 2)
    3. print(879 % 10)
  

<details>
<summary style="border-left:3px solid #3c763d; border-radius:2pt; width:100%; color:#3c763d; padding:6px; background-color: #dff0d8"> 
Réponses
</summary>  

<div style="border-left:3px solid #3c763d; border-radius:2pt; color:#3c763d; padding:6px; background-color: #eff0e8">

1. `1`<br>
2. `0`<br>
3. `9`<br>

    
</div>
</details>

### Afficher plusieurs valeurs dans un print

Il est aussi possible d'afficher plusieurs valeurs sur la même ligne avec la commande `print`. Pour ce faire, il suffit de séparer les valeur avec des virgules.
Par exemple, `print("Aujourd'hui, j'ai", 20 * 2 + 1, "ans")` affichera `Aujourd'hui, j'ai 41 ans`.

<h3 style="color:teal;background-color:azure;" > <i class="fa fa-pencil" aria-hidden="true"> </i> &nbsp; Exercice 9 </h3>

Ecrire un programme qui affiche `Le résultat de 10192 / 13 est :` où à la place de `10192 / 13`il y a le résultat du calcul . Vérifier en appuyant sur Ctrl + Enter.

In [ ]:
# à compléter


<h3 style="color:teal;background-color:azure;" > <i class="fa fa-pencil" aria-hidden="true"> </i> &nbsp; Exercice 10 </h3>

Ecrire un programme qui affiche un sapin
```
   X
  XXX
 XXXXX
XXXXXXX
   X
   X
```

In [ ]:
# à compléter


<h3 style="color:teal;background-color:azure;" > <i class="fa fa-pencil" aria-hidden="true"> </i> &nbsp; Exercice 11 </h3>

Ecrire un programme qui affiche les annimaux suivants
```
|\---/|
| o_o |
 \_^_/
```

```
     __
 .--()°'.'
'|, . ,'
 !_-(_\
```

*Voir sur [www.asciiart.eu](https://www.asciiart.eu/animals) d'autres idées*

In [ ]:
# à compléter

---

#### Remarque générale

Ce document a été élaboré par les enseignant du collège Sismondi à Genève. Il est sous license Creative Commons [BY-NC-SA](https://creativecommons.org/licenses/?lang=fr)
![Licence Creative Commons](https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png)
